### ISSUE TITLE
can't install latest deb

### ISSUE THREAD

**Issue by Tcll** [uTox/uTox#1323]
### Overview

1. uTox version: 0.17.0-1
2. Operating system: Xubuntu 16.04
3. Desktop Environment/Window Manager (Linux/BSD only): XFCE/xfwm4

### Issue

![utox_error](https://user-images.githubusercontent.com/773959/52527043-58e7a680-2c90-11e9-9062-c240e6703cc4.png)

just to elaborate on why I'm using 16.04 and not 18.04
I'd tried the upgrade, and there's quite a few things I don't like about it
1: it's slower
2: it uses more RAM
3: it's not as stable and a bunch of things don't work (eg the system load monitor no longer shows exact RAM/swap use when hovering)
4: tons of included software packages (eg gnome-software) are broken
basically, 16.04 just works, 18.04 doesn't

**Comment by GrayHatter** [comment]
What libfontconfig do you have?

And where's this .deb from?

**Comment by Tcll** [comment]
tried installing it since I couldn't seem to find it, but apparently I had it:
`libfontconfig1 is already the newest version (2.11.94-0ubuntu1.1).`
I guess ldconfig has issues :P
![check_libfontconfig1](https://user-images.githubusercontent.com/773959/52533210-689dd400-2cfe-11e9-8abe-c8ee7da6ed49.png)

anyways, the deb is from the releases page:
![utox_download](https://user-images.githubusercontent.com/773959/52533189-2e343700-2cfe-11e9-9541-311346e4b085.png)

**Comment by Tcll** [comment]
sorry this took a bit, I've been rather busy overall
I just went through and actually looked at the release page again and realized the deb wasn't actually there...
so I took a look at my download history to see where I got it from
looks like I got it from the debian archives and not the release page:
http://ftp.br.debian.org/debian/pool/main/u/utox/utox_0.17.0-1_amd64.deb
sorry about that

still, latest version though...
would this be a build issue on their end or a dependency issue on your end??

**Comment by Tcll** [comment]
any luck fixing this?? (at least provide an appimage if not a deb)
still stuck on the broken 0.16.1 appimage
- can't send files
- can't set ava (except by editing the file itself)
- can't set volume
- broken combo boxes
- random crashes

also had to create a new profile because it corrupted the previous one and I didn't have a backup :P

**Comment by avoidr** [comment]
We never provided packages for any distros, and never will. Please don't download binaries from random places on the internet. You're putting your friends at risk.

You need to use your distro's repository. If you choose to stay on an outdated system, you're on your own.

**Comment by Tcll** [comment]
you could at least build an appimage that would work on any distro and doesn't need to be installed

repositories aren't portable

**Comment by avoidr** [comment]
Listen. I told you multiple times already; there never was and never will be an "AppImage". At most, there will be a statically linked binary, but that's gonna happen on **my** whim, not yours. And not for a long time either, because I just have better things to do than that. No amount of complaining about it will change that — and you're about to spoil the idea of distributing binaries for me at all.

>you could at least build an appimage that would work on any distro and doesn't need to be installed

And you could just install uTox from your distro's repo.
You're audacious. You keep making demands to me like I owe you something, all while you refuse to type literally one single line into your terminal and install uTox like nature intended. And you've refused to type that fucking line for two years now.

>repositories aren't portable

You don't know what the fuck you're talking about.

>also, this page isn't some random website, which is where I got the 0.16.1 appimage from (no longer available).

Joke's on you, this very much **is** some random website. This domain has never belonged to uTox, we have no idea who is running that website! Our website was hosted under another domain, and the domain expired.

>this page is also official, and unlike utox, provides an easily downloadable appimage for qtox

I don't give a flying fuck what other projects do and what other people are doing in their spare time.

I told you before and I will say it one last time: It's not my job to build and distribute binaries or packages or anything like that. I am a developer and my job is to write code. This is what I am doing and I am doing that in my spare time. And anything extra that I do is on my whim.
Building and providing packages in distros is the job of distro packagers. They're also doing it in their spare time, for the most part. All of us are volunteers and we freely and independently do whatever we want and we're doing it the best we can. And you are walking around being an audacious prick who's making demands and dictating us how we are to spend our time. That's outrageous and plain rude. If you're going to come into some new community, you need to show some respect and do as the Romans do, or take your shit and go right back where you came from.

>at least this is better when you need a portable application.

You don't need a portable application. You need to sit your lazy punkass in front of a fucking terminal and type three words. You've been slacking off on it for two years already, and you need to do it right now.

>it's a shame you want to make people's lives harder by forcing them to use the repo.

You're being ridiculous.

>also it's not my fault life is keeping me on an outdated distro

And it's not my problem you can't manage your life to update your distro. I am not responsible for your life.

>if I could upgrade, I would,

You can.

>but upgrading would make my system unusable.

Bullshit.

>and I can't currently format and install Void either on this system.
>I'm just stuck and waiting for an opportunity to get out.

Ask yourself honestly when you think that "opportunity" will arise, when the world bows down to all your wishes and you always just manage to kinda sorta keep your outdated shit running. What incentive will you have to update? I can tell you; none. You will be slacking off on updating your system and you will keep running it for as long as possible. And you want to use Tox on your outdated, insecure system and you have the shamelessness to ask uTox for help? **I strongly object.** And I find it rude that you even have the audacity to ask such an irresponsible thing of us. **You're putting your friends at risk.** Please come to your senses and update! Or if you can't update, then please refrain from using uTox until you do.

>drop the elitism please

What elitism? You simply dialed the wrong number! And you keep harassing us! We are here to write code for uTox, nothing more, nothing less! What you expect of us is completely out of place! We are not responsible for your computer, operating system, your life, or your motivation issues. Those issues can be solved elsewhere. Not on the fucking uTox repo. Please seek help in appropriate places.

One more time, once and for all:
In the Linux world, we install software from official(!) distribution repositories. This makes sure that the software all comes from a quality-controlled source, and unmaintained parts of the system are reduced. The system is maintained and security issues are dealt with. Installing software from unofficial sources or making your own modifications to your system is at your own risk and is unsupported. No one is obliged to help you in that case, and it has nothing to do with elitism. Much less if you're using and outdated system.
If your desired software doesn't exist in the repo, you may kindly request it at your distribution.
If all that fails, you need to learn and do things yourself.
Refusing to learn, and demanding things from others, is considered rude.

In the Windows world it is reasonable to expect an exe straight from the developer. uTox does provide exes. The right place to download them from is the repo's releases page on GitHub. The files all come with their respective gpg signatures. It's your responsibility to verify the authenticity and integrity of the files. Failing to do so is at your own responsibility.

I believe this is all the information that you need for your journey.

Good luck.

**Comment by Tcll** [comment]
relax, I'm not trying to make demands, only suggestions... I'm just annoyed, that's all.

but if you want me to humor you, here ya go:
![Screenshot_2020-11-30_06-21-12](https://user-images.githubusercontent.com/773959/100604276-4f4f8880-32d4-11eb-82c0-eb1403aabf6e.png)
no different from before (and yes I've recently ran an `apt-get update`)
if I upgrade my OS, I'll keep the last bits I need this OS for when I can get my primary back online
but I'll use an extra ~200MiB of RAM for the OS alone, which will prevent me from opening the applications I normally use.
yes it's literally that critical that I don't have 200MiB to give to an OS upgrade.
(this thing is maxxed at 4GB, and Vivaldi has been starting to get greedy, and before you make the suggestion, firefox doesn't have audio, and uses even more RAM)

I never asked you to be responsible for my life, I'm just informing you what I currently can't do and why.
when I can build a secure router to get my primary back online, I can manage the last few things I need to do before I can format this OS and install Void.
but it all starts when I can get my hands on a machine capable of doing what I need (not just being a router)
until then, I'm stuck on an insecure outdated distro.

all I'm asking for is that you provide a cross-distro solution
if I can do it, I know you can

**Comment by avoidr** [comment]
> relax, I'm not trying to make demands, only suggestions... I'm just annoyed, that's all.

I'm also annoyed, sorry. I admit I went a little bit overboard, but I
honestly didn't know a better way to handle this anymore.

> but if you want me to humor you, here ya go:
> ![Screenshot_2020-11-30_06-21-12](https://user-images.githubusercontent.com/773959/100604276-4f4f8880-32d4-11eb-82c0-eb1403aabf6e.png)
> no different from before (and yes I've recently ran an `apt-get update`)

Yeah, this is where you have to compile by yourself. Because I suspect
that your distro is too old that they wouldn't add any more packages. So
requesting won't help. You can see if you can still request, and then
that is the right way.

If you can't request the package for your current distro version, then
the next right thing is to compile by yourself. The whole process and
requirements are documented in `docs/BUILD.md` and
`docs/DEPENDENCIES.md`. If you're having trouble with that process, it
likely means that the documentation is not clear enough and you can give
feedback and the documentation may be clarified.

It isn't as scary or elite as it may seem. You will learn a bit and
you'll feel nice about having your "own" uTox and your problem is dealt
with. Another nice thing is that you can always have the latest
development version and don't have to wait for releases or package
updates.  But please update your system as soon as possible.

> if I upgrade my OS, I'll keep the last bits I need this OS for when I can get my primary back online
> but I'll use an extra ~200MiB of RAM for the OS alone, which will prevent me from opening the applications I normally use.
> yes it's literally that critical that I don't have 200MiB to give to an OS upgrade.
> (this thing is maxxed at 4GB, and Vivaldi has been starting to get greedy, and before you make the suggestion, firefox doesn't have audio, and uses even more RAM)

I can very much relate, as I've also just upgraded my hardware this year
just to be able to view text documents. The quickest thing I can suggest
is finding some RAM somewhere. From an old computer, or ask friends. You
should be able to find something. I upgraded from 4 GiB to 8 GiB and it
helped a little.

> all I'm asking for is that you provide a cross-distro solution
> if I can do it, I know you can

You may also try installing via guix. It's a distro-independent package
manager. I believe the packages are also self-contained, so they work
independent of the system. But I have never used guix, you need to read
about it yourself.

An added benefit is that guix is maintained, so at least the packages
from guix shouldn't have security issues. So you can try guix in your
emergency situation, but it's still important that you update your
system as soon as possible.

Actually, now that I think about it, I think the guix way will be the
more appropriate thing to do in your situation. Because of the
maintained dependencies.

The uTox version in guix is currently outdated, but I may update that
package this week or so.
Guix is mentioned in docs/INSTALL.md

Hope that helps.

**Comment by Tcll** [comment]
thanks for toning down the mood, and I'm sorry for any added stress on your end.

I actually forgot about guix in my last message, but yeah it seems to be another flatpak solution, though more similar to npm in terms of use.

honestly though, compiling myself wouldn't make me feel good about having my own utox...
what would make me feel good is if I could build my own appimage to release for everyone else to use and have an updated version of utox with.
but I'm not sure how to properly build a system-independent appimage of a C program... (the errors you previously experienced with appimages)
my friend is actually still using the 0.16.1 appimage, and I'd like to provide her with an updated version.

when I said I could do it, I'm actually talking about an anaconda interpreter I've ripped to bits only with specific dependencies like PIL included.
all of it can be packed into an appimage to run independently on almost any version of linux.
so yeah, I can do it, but not with C yet unfortunately... u.u

btw, as for relating
when I said I was maxxed, I actually meant can't have any more than 2x2GB of DDR3 SO-DIMM on this potato Atom D2550
and it's because of that I'm actually restricted to lightweight distros, which Xubuntu no longer is
I'm just stuck between a rock and a hard place currently, and it won't be improving any time soon (unless a miracle happens).
this thing can't even run a VM of Void where I could install an updated utox from.

could you at the very least try to find someone who can properly build an appimage for people like me :/
even if I could install from the repo, I still prefer appimage because I can run it (when built properly) from a flash drive on any distro
you'd greatly contribute to the community there (this is what I'm focused on, making everyone else's lives better)
guix is more of a backburner solution if nobody's willing to make any moves.

the other perc to having an appimage (which I don't see this happening with utox any time soon) is the fact that new versions tend to break things that previously worked, like I've experienced with LBRY [here](https://github.com/lbryio/lbry-desktop/issues/4922)

**Comment by avoidr** [comment]
>look, I'd pay you money if I could for working on this

Better use that money for new hardware.

>but the current repo system isn't maintainable when you have at least 20 of them to support, and the maintainers aren't willing to stay up to date.

It becomes less maintainable when you disperse motivation to the likes of AppImage and Flatpak. Maintainers of distro repos update what people request, or what they want updated personally. If everyone starts sourcing packages from weird unofficial places, the security/immune system of the ecosystem becomes weak. It's like vaccinations in reverse. Please concentrate efforts on serious sources. Guix is a serious package manager with skilled people who know what they are doing. It isn't just random strangers. Guix is also not just a fad, it has a purpose and it has real goals and they're heavily invested in the project and to bring value, like reproducible builds, which are related to security. There is a whole distro (GuixSD) based on guix. The guix project is trustworthy.

And I'm actually about to submit an update to the uTox package in guix.

AppImage is not the solution to our problems. It's making the problem worse. It's just a bad workaround, and there is no end in the spiral of workarounds since then. Soon enough you'll need a whole stack of workarounds and everything will be in catastrophic state, which we may never recover from. The right thing to do is to stick to our repos and participate and do it right.

>it's up to the devs to ensure security, not the repo maintainers, unless it applies to the distro/OS itself

No, it's not. Developers sure have to make effort that *their code* is not faulty. But they can only control their own code. They can't control what you do with your computer. Your computer is your responsibility. And the security of your distro is the responsibility of the repo/distro maintainers. Please let's all stick to what we do best, and let's trust each other and work together.

>that's why snap, flatpak, and appimage exist after all

The reason why they exist is that someone 30 years or so ago was influential enough to start the "static linking is the enemy" religion. Of course that's bullshit, but those AppImage things are just a bad workaround.

>what would make me feel good is if I could build my own appimage to release for everyone else to use

No, please don't do that. Please don't propagate the idea that it's okay to run random binaries from random strangers. It undermines the security and trustworthiness of our systems in general.

>could you at the very least try to find someone who can properly build an appimage for people like me :/

I can't do that with good conscience.

>even if I could install from the repo, I still prefer appimage because I can run it (when built properly) from a flash drive on any distro

You can do that with guix, too, as far as I know.

>guix is more of a backburner solution if nobody's willing to make any moves.

Disagree. Guix is a solid and trustworthy system.

>the other perc to having an appimage (which I don't see this happening with utox any time soon) is the fact that new versions tend to break things that previously worked, like I've experienced with LBRY here, so I could keep the older appimage around until the issue is fixed in a much newer version

Well, that shouldn't be an issue with uTox :^) For that situation, compiling from source is the best way, because you don't need to wait at all. You can install multiple versions of a package side-by-side in guix. Try it.

**Comment by Tcll** [comment]
heh, I'm glad to see the same treatment I give others come back around to me for once...
while yes I do need the money more for better hardware, I'd still give out some anyways for motivation, and even more so now for that respect :+1: 

but anyways...
> Maintainers of distro repos update what people request

I have yet to really see that take place with Void (I see some, but they could do better)
> or what they want updated personally.

yeah that's more like it
> the security/immune system of the ecosystem becomes weak.

as if it was good to begin with, users are more inclined to follow a random PPA from some random github page because the distro doesn't add it by default until 3 upgrades later, unless there's a clear warning about it being untrusted.
appimages are literally just as secure in that sense, you're best to just use common sense as with every other system.

> And I'm actually about to submit an update to the uTox package in guix.

please

> The right thing to do is to stick to our repos and participate and do it right.

the last things that suck about this and why I hate it is because this is really only good for OS/distro related stuff like themes
gparted would be better as an appimage
but that's not the real reason, the real reason comes from when you multi-boot different distros for different purposes when you need something (which I do on my primary)
it's better to have a shared Programs directory on another hard drive linked between them like I do.

best security for this is user-sided common sense, but I should be open to that freedom, not be forced to install the same program multiple times because the walled garden dictates and controls what I can do, especially when the OS corrupts itself, or the OS just gets too messy where a reinstallation is just preferred.
I don't mind setting up numerous launchers on my makeshift mac-bar every time I reinstall (not often)
all I need to do is symlink my config directories and launch the program from my other hard drive (portability)
this is why the repo isn't portable to users

> Please let's all stick to what we do best, and let's trust each other and work together.

> No, please don't do that. Please don't propagate the idea that it's okay to run random binaries from random strangers. It undermines the security and trustworthiness of our systems in general.

you realize you've just contradicted yourself here right??
(no you're not some random foreigner on a blogger page building the app, but you're still a random stranger)
trustworthiness is up to the user to decide things for themselves, and also up to the developer to actually appear trustworthy, not the maintainer of a walled garden (I'm very much against walled gardens, especially with what Google, Microsoft, Apple, and even Canonical (least offender) have displayed)

it's best if the distro focuses on the distro and not all the apps the user can install on it (Win8).
besides, security is already defeated when the maintainers of the walled garden throw something like Discord in their repo
may as well download an appimage from discord themselves which is just as trustworthy.
users should have freedom to do what they want, and that's what appimage provides
portability is also freedom, which repos don't provide. (can't install a deb on Void Linux, unless the package is portable enough for it to be run with dpkg)
Solus at least had a decent idea...

but I'm sorry, I can't agree here on that when I can't run (let alone obtain) half the programs I want to
especially when I have to build from source and do the job of the developer on my end
appimage takes away that burden from me because I can download and run on any distro if it's built properly.
(yes building from source is a burden, so much so in fact it keeps my friend on Windows, just look at GIMP and try to tell users to build that, it's a nightmare)
it's not up to the user to build from source, though it should at least be an option at minimum if THEY want to (which is also freedom)

again, I'll be happy when users can just download and run as they expect with minimal to no effort on their part
btw, you're ignoring how appimage provides a sandbox for security ;)

> Disagree. Guix is a solid and trustworthy system.

so is appimage, not so much flatpak and definitely not snap
but yeah, it's on the backburner for me, mainly because it appears it's basically where flatpak was at
I would like to push for something that doesn't have to be installed with excessive HDD bloat first.
(as well as RAM bloat from constantly running a daemon or server)

> It becomes less maintainable when you disperse motivation

this inherently proves it's flawed
if it becomes a hassle when you have to manage numerous build systems for each distro then a more portable solution should be sought.
which is actually another thing appimage actually solves because it works across all distros and doesn't really need much special attention.
(not sure how much applies to C, but at least appimage supports static linking with squashfs to back it)

> For that situation, compiling from source is the best way

highly disagree there, because it's usually a hassle to build from source (that's generally speaking, not for utox which has solved a lot of it)
what I did with LBRY (excluding the sh and forced updates) is actually the best way as the user has complete control
I'll back guix on this (and I believe flatpak does this too) but it's not as desirable as you need a special interface to manage things, where as appimage only needs your own directory structure how you want it.
^ convenience interfaces are just as much of a problem as they are a feature (unless you go the route of Mint's package manager)

I'm not against a package manager
I'm against the enforcement of a package manager for everything
(basically Microsoft back in 2012 with Win8)

**Comment by avoidr** [comment]
> > the security/immune system of the ecosystem becomes weak.
> 
> as if it was good to begin with,

It *was*. But then some enthusiasts started pushing "Linux" for "the
year of the Linux desktop".

> users are more inclined to follow a random PPA from some random github page because the distro doesn't add it by default until 3 upgrades later, unless there's a clear warning about it being untrusted.

That's the recent development, and I ain't takin' no part in it.

> appimages are literally just as secure in that sense, you're best to just use common sense as with every other system.

Common sense is not that common anymore, sadly.

> 
> > And I'm actually about to submit an update to the uTox package in guix.
> 
> please

Here you go:
https://lists.gnu.org/archive/html/guix-patches/2020-11/msg01095.html

> 
> > The right thing to do is to stick to our repos and participate and do it right.
> 
> the last things that suck about this and why I hate it is because this is really only good for OS/distro related stuff like themes
> gparted would be better as an appimage
> but that's not the real reason, the real reason comes from when you multi-boot different distros for different purposes when you need something (which I do on my primary)
> it's better to have a shared Programs directory on another hard drive linked between them like I do.

Yeah, I understand that situation. But I think I personally wouldn't try
to mix systems like that, and only share data (/home), and only after
thinking about possible conflicts between program version mismatches (I
would be very careful with dotfiles).

It just doesn't sound very pretty to me to have some out-of-distro,
independent system. It's hard to keep updated, it's just messy. I'd
probably prefer to maintain two installations as independent systems.

But these are just really quick thoughts of mine, I never really had
that problem, so I don't know what best solution I would come up for
myself.

It's quite possible that I'd actually look at guix as a possible
solution, since it's distro independent and portable.

> best security for this is user-sided common sense,

Yeah, which becomes less and less common. Think about it; "everyone
knows" `curl | sudo sh` is bad. But we're all running huge — and I mean
**huge** — remote code execution engines (of more or less dubious
quality) as our user, which has access to *all* our personal data.
Virtually no one questions that.

And then we *still* have sudo to be the cherry on top.

> but I should be open to that freedom, not be forced to install the same program multiple times because the walled garden dictates and controls what I can do, especially when the OS corrupts itself, or the OS just gets too messy where a reinstallation is just preferred.
> I don't mind setting up numerous launchers on my makeshift mac-bar every time I reinstall (not often)
> all I need to do is symlink my config directories and launch the program from my other hard drive (portability)

Oh, so you use symlinks for dotfiles. Yeah, you could control more
granularly which dotfiles you want to share.

> this is why the repo isn't portable to users
> 
> > Please let's all stick to what we do best, and let's trust each other and work together.
> 
> > No, please don't do that. Please don't propagate the idea that it's okay to run random binaries from random strangers. It undermines the security and trustworthiness of our systems in general.
> 
> you realize you've just contradicted yourself here right??
> (no you're not some random foreigner on a blogger page building the app, but you're still a random stranger)
> trustworthiness is up to the user to decide things for themselves, and also up to the developer to actually appear trustworthy, not the maintainer of a walled garden (I'm very much against walled gardens, especially with what Google, Microsoft, Apple, and even Canonical (least offender) have displayed)

By "trust each other" I mean to trust our competence, instead of
everybody becoming a poor full stack engineer and making effort
redundant.

Sure, we are all random strangers. But the software developer who
publishes his code or possibly even binaries, he has a face. And the
distro maintainers and packagers, they have a face. But the random
stranger uploading a binary to some random website, he doesn't have a
face. (By the way, this is where reproducible builds come into play —
the build process is documented and reproducible, so the source of a
binary doesn't need to be trusted, because the binary can be verified.
You don't even need to trust the original author of the software. Guix
can also do reproducible builds.)

> it's best if the distro focuses on the distro and not all the apps the user can install on it (Win8).
> besides, security is already defeated when the maintainers of the walled garden throw something like Discord in their repo
> may as well download an appimage from discord themselves which is just as trustworthy.

Yes, I also condemn that practice for multiple reasons. But as long as
distros don't install them by default, I'm fine with that. And still, at least
they're creating the packages straight from the original author and not
some completely random stranger. So at least you can trust the *source*,
even if the original author is not trustworthy.

I think the distro should focus on integration. It's hard to pull
software (packages, binaries, …) from all kinds of sources and expect a
reasonable, non-messy result. If it works together at all. They should
focus on security, but they can't just accuse someone of being
malicious. There are different needs of security and it's what every
user's job to judge for himself who is trustworthy. After all, there is
even some free software that I don't trust even though I could look at
the code.

> users should have freedom to do what they want, and that's what appimage provides
> portability is also freedom, which repos don't provide. (can't install a deb on Void Linux, unless the package is portable enough for it to be run with dpkg)
> Solus at least had a decent idea...
> 
> but I'm sorry, I can't agree here on that when I can't run (let alone obtain) half the programs I want to
> especially when I have to build from source and do the job of the developer on my end
> appimage takes away that burden from me because I can download and run on any distro if it's built properly.
> (yes building from source is a burden, so much so in fact it keeps my friend on Windows, just look at GIMP and try to tell users to build that, it's a nightmare)
> it's not up to the user to build from source, though it should at least be an option at minimum if THEY want to (which is also freedom)

I'm really curious what software you need that isn't packaged.

> again, I'll be happy when users can just download and run as they expect with minimal to no effort on their part
> btw, you're ignoring how appimage provides a sandbox for security ;)

Yeah, Flatpak also provided a sandbox…

> 
> > Disagree. Guix is a solid and trustworthy system.
> 
> so is appimage, not so much flatpak and definitely not snap

Definitely not all of the three. Seriously, they can't even be compared
when it comes to "solidness" and trustworthiness.

> but yeah, it's on the backburner for me,

That's too bad.

> mainly because it appears it's basically where flatpak was at

Where was Flatpak at?

> I would like to push for something that doesn't have to be installed with excessive HDD bloat first.
> (as well as RAM bloat from constantly running a daemon or server)

I don't know how installing excessive HDD bloat (and RAM bloat from
keeping each library from each AppImage in RAM) afterwards is better.

AppImage is not a solution to any problem, it is a problem itself, and
causes further problems, and consequences of which I can only warn
about.

> 
> > It becomes less maintainable when you disperse motivation
> 
> this inherently proves it's flawed

By that statement *everything* is flawed. If everyone does everything
for himself, there is tremendous waste of effort and lower quality. Only
by working together you can build something bigger and better.

> if it becomes a hassle when you have to manage numerous build systems for each distro then a more portable solution should be sought.

What numerous build systems?

> which is actually another thing appimage actually solves because it works across all distros and doesn't really need much special attention.
> (not sure how much applies to C, but at least appimage supports static linking with squashfs to back it)

What do you mean by "static linking" here?

> 
> > For that situation, compiling from source is the best way
> 
> highly disagree there, because it's usually a hassle to build from source (that's generally speaking, not for utox which has solved a lot of it)
> what I did with LBRY (excluding the sh and forced updates) is actually the best way as the user has complete control
> I'll back guix on this (and I believe flatpak does this too) but it's not as desirable as you need a special interface to manage things, where as appimage only needs your own directory structure how you want it.
> ^ convenience interfaces are just as much of a problem as they are a feature (unless you go the route of Mint's package manager)

Well, you can't avoid around learning something if you want to have
something that is useful to you. That's true of all areas of life.

> I'm not against a package manager
> I'm against the enforcement of a package manager for everything
> (basically Microsoft back in 2012 with Win8)

A package manager enforces nothing. You're free to install software
whatever way you want. But AppImage is a bad choice for that. It may be
really regrettable, but if you want to have a fun life, then you need to
learn *something* to some degree. There is no other way.

**Comment by Tcll** [comment]
> Well, you can't avoid around learning something if you want to have
something that is useful to you. That's true of all areas of life.

why should I have to learn to build what I want to use?
why can't I just have what I want to use already available for me in an easy package that runs everywhere I need it to so I only have to learn what I want to use rather than how to build it?

> but if you want to have a fun life, then you need to
learn *something* to some degree. There is no other way.

this is counter-intuitive when I'm forced to learn something I shouldn't have to learn just to use something I want to have fun with.
and that isn't fun, it's a burden.

and a package manager only makes things worse when they refuse to get off their lazy butts and actually add such useful packages.
it's hard to trust competence when it's not shown.

> By that statement *everything* is flawed. If everyone does everything
for himself, there is tremendous waste of effort and lower quality. Only
by working together you can build something bigger and better.

I mean... yes and no... >_>
I've had to resort to stripping out the standard library of a portable anaconda interpreter just so I could have a competent universal solution everyone could run on their machine, and also back with a portable appimage
(what isn't flawed is what makes everyone's lives easier while keeping them secure)

> I don't know how installing excessive HDD bloat (and RAM bloat from
keeping each library from each AppImage in RAM) afterwards is better.

>AppImage is not a solution to any problem, it is a problem itself, and
causes further problems, and consequences of which I can only warn
about.

sadly it's the solution to incompetence because people can't build software that both runs on everything and is easily available on everything.

I'll agree with you on the RAM bloat, but that's a sacrifice I'm willing to make because it at least doesn't hog RAM long after closing the process. (1 reason I don't like be-all end-all solutions like guix and flatpak)

HDD bloat however, appimage is actually better at, because while yes you have copies of numerous versions of libraries all **statically linked to their local sandbox** (regarding your question), that's still much less bloat than what comes with be-all end-all solutions like guix and flatpak.
(I'll defend guix though since you mentioned you can copy your installations to a flash drive to other distros like appimage)

> I'm really curious what software you need that isn't packaged.

one example here would be gimp 2.8.3 because it was the most stable version I've ever used that's actually worked properly.
(sadly I don't even have that anymore because it's been patched to 2.8.16, except for maybe in one of my OS backups, but it's not portable for me to simply grab and use)
but the lowest I can get that's actually usable is 2.9.7, which is full of bugs, while everything higher is completely broken.

I kinda forgot what other software I wanted because I've found alternatives that worked decently enough.

and btw, regarding earlier statements, first of all I want to apologize for the initial wrong idea.
but the only thing that advocates random strangers posting binaries is when developers don't provide them.
which even without appimage, there's still app.tar.gz like Blender and JetBrains do.
(I should be able to go to your releases page (github/gitlab/website) and download a binary (appimage/standalone/archive) at my own leasure)
^ I don't have to build Blender or CLion just to use it

I shouldn't be forced to install random bloat like guix, flatpak, app-outlet or what have you just to use what I want to use...
yes freedom is a double-edged sword, but the best we can do is simply warn users about what's bad and let them decide for themselves.
if they choose to get destroyed on Discord, Google, GitHub, or what have you, with all of their private stuff made public or owned by their slavemasters after clearly being warned, then that's on them.
don't hamper the smart just because the stupid wish to ignore, and don't divide the smart from the stupid by restricting software to the smart.
(staying up-to-date isn't always the smart move, just look at windows which has gotten less secure (over all) since the XP days)
^ and in my case, not currently having a choice doesn't make me stupid (all of my sweets are stored offline).

but hey, I'm glad you provided a guix package so people who use it may benefit from it.
the first thing I think about is everyone else, which is why I'm suggesting so hard you release an appimage on your releases page.
(I don't care if you want microsoft to own and dictate your code by using this platform instead of gitlab, it's your freedom to do what you want)
^ and that includes the final result if you ultimately decide not to release your own binary (*.tar.gz/*.appimage)


and finally for the last thing to cover:
> What numerous build systems?

that would be what you define to quickly build and/or package your binaries (which I can tell you haven't done)
I'm not sure what things like LBRY, Minetest, and KDevelop use for that as I don't use build systems myself (it's inefficient for me to manage myself).
(I prefer to build the binary myself rather than through numerous automated systems and have confidence by using what works for everyone else)
^ this likely means I'll probably never release a package (only an appimage at minumum), but that doesn't really matter since I'll make my binary easily obtainable (trust it from my release, not from some foreigner's blogger page).

though that doesn't mean distro managers can't package my binary for their distro (which is also trusted) if they want to or are requested to.
they're free to collaborate with me, or rely on release notifications.
(I'm stuck between a rock and a hard place for that to happen any time soon for my software, plus that's still WIP)
^ building the software I want to use causes delay here.

**Comment by avoidr** [comment]
> > Well, you can't avoid around learning something if you want to have
> something that is useful to you. That's true of all areas of life.
> 
> why should I have to learn to build what I want to use?

Because you want to use it. And you don't want to rely on other people.
Why should you have to cook something that you want to eat?

> why can't I just have
> what I want to use
> already available for me
> in an easy package
> that runs everywhere I need it to
> so I only have to learn what I want to use rather than how to build it?

I'll just leave it here like this.

> > but if you want to have a fun life, then you need to
> >learn *something* to some degree. There is no other way.
> 
> this is counter-intuitive when I'm forced to learn something I shouldn't have to learn just to use something I want to have fun with.
> and that isn't fun, it's a burden.

Then go use something that doesn't require effort and learning. Then see
how much fun you can have.
And I don't mean the kind of fun that you mean. Why should I use my time
to make your life fun? Don't make me angry now.

> and a package manager only makes things worse when they refuse to get off their lazy butts and actually add such useful packages.

It doesn't work like this here, kid. Maybe you should have some respect
for the work people are doing.

> it's hard to trust competence when it's not shown.

Maybe it's not shown because there's such thing as *priorities*. Because
maybe people aren't 12 years old with lots of time to throw out and no
responsibilities.

Besides that, whom do you think competent people want to impress?

> (what isn't flawed is what makes everyone's lives easier

So far, AppImage has only made my life harder.

> while keeping them secure)

Objectively false, and you're proof of that. Fucking holding on to
years-outdated OS just because "secure" binaries from fuck-knows-where
enable you to.

> >AppImage is not a solution to any problem, it is a problem itself, and
> >causes further problems, and consequences of which I can only warn
> >about.
> 
> sadly it's the solution to incompetence because people can't build software that both runs on everything and is easily available on everything.

Go do it yourself then. Show your competence. Start by copypasting lines
from `docs/BUILD.md`.

> HDD bloat however, appimage is actually better at, because while yes you have copies of numerous versions of libraries all **statically linked to their local sandbox** (regarding your question),

Maybe you should read what static linking actually is, because I'm
pretty sure it's not what AppImage does. If it did, it would obsolete
itself.
And it doesn't answer my question.

> (I'll defend guix though since you mentioned you can copy your installations to a flash drive to other distros like appimage)

I mentioned. You should have about it all by yourself by now. I thought
you're stuck in such a hard place.

> > I'm really curious what software you need that isn't packaged.
> 
> one example here would be gimp 2.8.3 because it was the most stable version I've ever used that's actually worked properly.
> (sadly I don't even have that anymore because it's been patched to 2.8.16, except for maybe in one of my OS backups, but it's not portable for me to simply grab and use)
> but the lowest I can get that's actually usable is 2.9.7, which is full of bugs, while everything higher is completely broken.

Aha. So you have some favorite version of a program but the world moved
on, you have all the tools that you need to produce any version you
like, and you figured that the best thing to do in your situation is to
go ask people to do it for you?

> I kinda forgot what other software I wanted because I've found alternatives that worked decently enough.

Lucky those, whose software you forgot.

> but the only thing that advocates random strangers posting binaries is when developers don't provide them.

No. Fact: The job of a developer is not to post binaries.

> (I should be able to go to your releases page (github/gitlab/website) and download a binary (appimage/standalone/archive)
> at my own leasure)

No. Go fuck yourself. You're not the center of the fucking universe,
nobody lives to please your lazy fucking ass. Fuck you and fuck your
leasure.

> ^ I don't have to build Blender or CLion just to use it

You don't have to use anything. Use something else. Go away and use
something else already. Elsewhere.

> I shouldn't be forced to install random bloat like guix, flatpak, app-outlet or what have you just to use what I want to use...

You're making me fucking angry. You've put yourself in your position,
now fucking deal with it.

> yes freedom is a double-edged sword, but the best we can do is simply warn users about what's bad and let them decide for themselves.

Yes, that's surely gonna work. That must be why over a million people
are still alive in the western world this year. That must also be why
you've *decided and taken your shit elsewhere*. *OH WAIT*.

> don't hamper the smart just because the stupid wish to ignore, and don't divide the smart from the stupid by restricting software to the smart.

Nice speech. But wait, which are you now? The hampered smart, or the
smart to whom the software was restricted to?

> the first thing I think about is everyone else,

Yes, I can see that.

> which is why I'm suggesting so hard you release an appimage on your releases page.

That's not helping anybody.

> (I don't care if you want microsoft to own and dictate your code by using this platform instead of gitlab, it's your freedom to do what you want)

Microsoft doesn't dictate anything. The code is free and has been cloned
180 times as of now.

> ^ and that includes the final result if you ultimately decide not to release your own binary (*.tar.gz/*.appimage)

I have ultimately decided that months ago, it was never a question. I
would never do such a stupid thing.
My binary is mine. Go make your own.

Elsewhere.
